conans/model/info.py

```diff
@@ -272,6 +280,9 @@ def package_id(self):
             return computed_id
         result = []
         result.append(self.settings.sha)
+        # Only are valid requires for OPtions those Non-Dev who are still in requires
+
+        self.options.filter_used(self.requires.pkg_names)
```

Comment:
It will only generate a new sha for those packages which ``package_id()`` is removing requires from it. I'd say that nobody is doing it, and if they are, it would be a broken behavior (add a new require, and get a new sha/ID). Better fix it now?